File: src/main/java/dev/lampirg/consultationappointment/service/student/AppointmentMaker.java
package dev.lampirg.consultationappointment.service.student;

import dev.lampirg.consultationappointment.data.student.Student;
import dev.lampirg.consultationappointment.data.teacher.DatePeriod;
import dev.lampirg.consultationappointment.data.teacher.Teacher;

public interface AppointmentMaker {
    void makeAppointment(Teacher teacher, Student student, DatePeriod datePeriod);
    void deleteAppointmentById(Long id);
    boolean isAvailable(DatePeriod datePeriod);

    default boolean isAvailable(Teacher teacher, Student student, DatePeriod datePeriod) {
        return isAvailable(datePeriod);
    }
}

File: src/main/kotlin/dev/lampirg/consultationappointment/service/student/AppointmentMaker.kt
package dev.lampirg.consultationappointment.service.student

import dev.lampirg.consultationappointment.data.student.Student
import dev.lampirg.consultationappointment.data.teacher.DatePeriod
import dev.lampirg.consultationappointment.data.teacher.Teacher

interface AppointmentMaker {
    fun makeAppointment(teacher: Teacher, student: Student, datePeriod: DatePeriod)
    fun deleteAppointmentById(id: Long)
    fun isAvailable(datePeriod: DatePeriod): Boolean
    fun isAvailable(teacher: Teacher, student: Student, datePeriod: DatePeriod): Boolean = isAvailable(datePeriod)
}